spec/support/test_app.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'rack/auth/basic'
+
+helpers do
+  def protected!
+    return if authorized?
+
+    response['WWW-Authenticate'] = %(Basic realm="Restricted Area")
+    throw(:halt, [401, "Unauthorized access\nYou are denied access to this resource."])
+  end
+
+  def authorized?
+    @auth ||= Rack::Auth::Basic::Request.new(request.env)
+    @auth.provided? && @auth.basic? && @auth.credentials && @auth.credentials == %w[guest guest]
+  end
+end
+
+get '/' do
+  "I'm Feeling Grovery"
+end
+
+get '/cookie_renderer' do
+  cookie_info = request.cookies.map.with_index(1) do |(name, value), i|
+    "#{i}. #{name} #{value}"
```

Comment:
not quite the same rendering as the original, but it'll do (the original used a table)

Reply:
Since I wasn't able to get in touch with the original, I had to resort to improvisation.

Reply:
No worries.. I wrote it, I've just "misplaced" the source.. What you did is fine 😄